dallinger/db.py

```diff
@@ -100,3 +102,26 @@ def wrapper(*args, **kw):
                 else:
                     raise
     return wrapper
+
+
+# Reset outbox when session begins
+@event.listens_for(Session, 'after_begin')
+def after_begin(session, transaction, connection):
+    session.info['outbox'] = []
+
+
+# Reset outbox after rollback
+@event.listens_for(Session, 'after_soft_rollback')
+def after_soft_rollback(session, previous_transaction):
+    session.info['outbox'] = []
+
+
+# Publish messages to redis after commit
+@event.listens_for(Session, 'after_commit')
```

Comment:
Do we know that the commit being handled is the one that's adding the Participant? (There's no explicit check, and no declarative interface/type registration so I'm not sure how that works. Maybe adding Participants is the only thing triggering database commits once the experiment is in progress?)